### PR TITLE
notifier bug fix and cleanups, deprecate non-batch format

### PR DESCRIPTION
**Pull request by Dieterbe**
* deprecate PersistMessage and remove parsing of that format.
  - NSQ uses the PersistMessageBatchV1 format since
    72d2f6e531bf3f8dd3b0c7efc0bd009192f3588e (jan 06 2016)
  - Kafka uses it since it was introduced
    02a8a92badee44899fd7c0eaecacd8afb5ca8b0c (jul 18 2016)

* make sure SavedChunk.Name is always properly set, even in
  mixed version scenarios

* simply notifier handling code, just make it utility function

* re-arrange field order to make more sense

* NotifierNSQ CreateMissing now defaults to true, like Kafka

### CODE REVIEW

File: mdata/notifier.go [hunk 1]
```diff
@@ -84,16 +70,26 @@ func (cl Notifier) Handle(data []byte) {
...
 				aggId, _ := MatchAgg(c.Name)
-				agg := cl.Metrics.GetOrCreate(key[0], c.Name, schemaId, aggId)
+				agg := metrics.GetOrCreate(key[0], c.Name, schemaId, aggId)
 				if len(key) == 3 {
 					agg.(*AggMetric).SyncAggregatedChunkSaveState(c.T0, consolidator, uint32(aggSpan))
 				} else {
 					agg.(*AggMetric).SyncChunkSaveState(c.T0)
 				}
-			} else if agg, ok := cl.Metrics.Get(key[0]); ok {
+			} else if agg, ok := metrics.Get(key[0]); ok {
```
Comment (replay) [inline mdata/notifier.go]: not really part of this PR, but line `87` - `97` seem like they could be refactored to reduce duplication like (if `agg` is declared at one larger scope):
```
			} else if agg, ok := metrics.Get(key[0]); !ok {
				continue
			}
			if len(key) == 3 {
				agg.(*AggMetric).SyncAggregatedChunkSaveState(c.T0, consolidator, uint32(aggSpan))
			} else {
				agg.(*AggMetric).SyncChunkSaveState(c.T0)
			}
```

File: mdata/notifierNsq/notifierNsq.go [hunk 2]
```diff
@@ -52,13 +53,11 @@ func New(instance string, metrics mdata.Metrics, idx idx.MetricIndex) *NotifierN
...
-		Notifier: mdata.Notifier{
-			Instance: instance,
-			Metrics:  metrics,
-		},
-		idx: idx,
+		instance:      instance,
+		in:            make(chan mdata.SavedChunk),
+		metrics:       metrics,
+		idx:           idx,
+		createMissing: true,
```
Comment (woodsaj) [inline mdata/notifierNsq/notifierNsq.go]: if both notifierKafka and notifierNSQ are hard coding createMissing=true, then we should just remove it.

**Comment by woodsaj** [comment]
replaced by PR #588
